### ISSUE TITLE
Issue with index created on large set of existing dates

### ISSUE THREAD

**Issue by donl** [appy-one/acebase#117]
I seem to have an issue when creating an index on a large set of existing date values.

Hopefully the code below is helpful.  Sorry for any bad practices :)

In some instances on subsequent executions (without deleting the db) I get an assertion error: `Assertion failed: Node entries are not sorted ok`

```js

const { AceBase } = require("acebase");

const dbname = "idx-test";
const options = {logLevel: 'error'};

let db = {};

const main = async () => {

    db = new AceBase(dbname, options);
    await db.ready();

    // Works better if index is created first...
    // await db.indexes.create('item','at');

    const preCount = await db.ref('item').count();
    console.log('preCount',preCount);

    var currentDate = new Date('1/1/2000');
    var stopDate = new Date('12/31/2022');
    var dates = {};

    while (currentDate <= stopDate) {
        const ref = db.ref('item').push();
        dates[ref.key] = {at: new Date (currentDate)};

        let d = new Date(currentDate);
        currentDate = d.setDate(d.getDate()+1);
    }

    await db.ref('item').update(dates);

    const postCount = await db.ref('item').count();
    console.log('postCount',postCount);


    snaps = await db.query('item')
    .filter('at','between',[new Date('5/1/2022'), new Date('1/1/2022')])
    .get();
    console.log("query 1 length: ",snaps.getValues().length);


    // problem with index created on bulk existing data
    await db.indexes.create('item','at').then(()=>{
        console.log('index created...')
    });

    snaps = await db.query('item')
    .filter('at','between',[new Date('5/1/2022'), new Date('1/1/2022')])
    .get();
    console.log("query 2 length: ",snaps.getValues().length);
}

main();

```

**Comment by appy-one** [comment]
I worked quite a bit on the indexes lately. Can you check if you are using the latest version?

**Comment by donl** [comment]
Running `1.21.4`

I see there is `api.getServerInfo()` that returns the server version.  Is there a good way to get the AceBase and/or client version?

**Comment by appy-one** [comment]
I'll investigate what's going wrong

**Comment by appy-one** [comment]
I found the issue, published a fix in v1.21.5. Thanks for the great example code, that saved a lot of work!

**Comment by appy-one** [comment]
> I see there is `api.getServerInfo()` that returns the server version. Is there a good way to get the AceBase and/or AceBase-client version?

Not yet, I'll implement that 👍🏼
